unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -7275,6 +7223,110 @@ ProceduralAsset asset
+
+            BoxCollider nonTriggeredBoxCollider = noneTriggeredEncapsulatingBox.AddComponent<BoxCollider>();
+            nonTriggeredBoxCollider.size = colliderSize; // Scale the box to the agent's size
+            nonTriggeredBoxCollider.enabled = true;
+
+            noneTriggeredEncapsulatingBox.transform.parent = agent.transform;
+
+            GameObject triggeredEncapsulatingBox = new GameObject("triggeredEncapsulatingBox");
+            // triggeredEncapsulatingBox.transform.position = new Vector3(bounds.center.x, bounds.center.y, agent.transform.position.z);
+            triggeredEncapsulatingBox.transform.position = new Vector3(bounds.center.x, bounds.center.y, bounds.center.z);
+
+            BoxCollider triggeredBoxCollider = triggeredEncapsulatingBox.AddComponent<BoxCollider>();
+            triggeredBoxCollider.size = colliderSize; // Scale the box to the agent's size
+            triggeredBoxCollider.enabled = true;
+            triggeredBoxCollider.isTrigger = true;
+
+            triggeredEncapsulatingBox.transform.parent = agent.transform;
+
+            if (useVisibleColliderBase){
+                if (useAbsoluteSize) {
+                    colliderSize = new Vector3(
+                        scaleRatio.x,
+                        0.15f,
+                        scaleRatio.z
+                    );
+                } else {
+                    colliderSize = new Vector3(
+                        scaleRatio.x * bounds.extents.x * 2,
+                        0.15f,
+                        scaleRatio.z * bounds.extents.z * 2
+                    );
+                }
+                GameObject visibleBox = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                visibleBox.name = "VisibleBox";
+                visibleBox.transform.position = new Vector3(bounds.center.x, 0.05f, bounds.center.z);
+                visibleBox.transform.localScale = colliderSize;
+                visibleBox.transform.parent = agent.transform;
+            }
+
+            this.transform.rotation = originalRotation;
+        }
+
+        public void DestroyAgentBoxCollider(ServerAction action){
+            GameObject nonTriggeredEncapsulatingBox = GameObject.Find("NonTriggeredEncapsulatingBox");
+            GameObject triggeredEncapsulatingBox = GameObject.Find("triggeredEncapsulatingBox");
+            GameObject visibleBox = GameObject.Find("VisibleBox");
+            if (nonTriggeredEncapsulatingBox != null) {
+                GameObject.Destroy(nonTriggeredEncapsulatingBox);
+            }
+            if (triggeredEncapsulatingBox != null) {
+                GameObject.Destroy(triggeredEncapsulatingBox);
+            }
+            if (visibleBox != null) {
+                GameObject.Destroy(visibleBox);
+            }
+            actionFinished(true);
+            return;
+        }
+
+        public void UpdateAgentBoxCollider(ServerAction action) {
```

Comment:
Also here I would replace the ServerAction action parameter into the more explicit syntax, so something like 

```public void UpdateAgentBoxCollider(Vector3 colliderScaleRatio, bool useAbosluteSize, bool useVisibleColliderBase)```

Similarly I would remove these added members of the ServerAction class as to not bloat it even more since we are trying to move away from having to use it.